src/gui/texture.rs

```diff
@@ -180,11 +107,45 @@ impl Texture {
         // TODO(cohae): add a method to GameVersion to check for prebl
         let is_prebl = matches!(
             package_manager().version,
-            destiny_pkg::GameVersion::Destiny2Beta | destiny_pkg::GameVersion::Destiny2Shadowkeep
+            GameVersion::Destiny2Beta
+                | GameVersion::Destiny2Forsaken
+                | GameVersion::Destiny2Shadowkeep
         );
 
         let mut cur = std::io::Cursor::new(header_data);
-        let texture: TextureHeader = cur.read_le_args((is_prebl,))?;
+        let texture: TextureHeaderGeneric = match package_manager().platform {
+            PackagePlatform::PS4 => {
+                let texheader: TextureHeaderD2Ps4 = cur.read_le_args((is_prebl,))?;
+                TextureHeaderGeneric {
+                    data_size: texheader.data_size,
+                    format: texheader.format.to_wgpu()?,
+                    width: texheader.width,
+                    height: texheader.height,
+                    depth: texheader.depth,
+                    array_size: texheader.array_size,
+                    large_buffer: texheader.large_buffer,
+
+                    deswizzle: Some((texheader.flags1 & 0xc00) != 0x400),
+                    psformat: Some(texheader.format),
+                }
+            }
+            PackagePlatform::Win64 => {
+                let texheader: TextureHeaderPC = cur.read_le_args((is_prebl,))?;
+                TextureHeaderGeneric {
+                    data_size: texheader.data_size,
+                    format: texheader.format.to_wgpu()?,
+                    width: texheader.width,
+                    height: texheader.height,
+                    depth: texheader.depth,
+                    array_size: texheader.array_size,
+                    large_buffer: texheader.large_buffer,
+
+                    deswizzle: None,
+                    psformat: None,
+                }
```

Comment:
Putting these conversions on a `From<T> for TextureHeaderGeneric` implementation on each header would make this code more reusable.
eg:
```rs
impl From<TextureHeaderD2Ps4> for TextureHeaderGeneric {
    fn from(/* ... */) -> Self {
        TextureHeaderGeneric {
            data_size: texheader.data_size,
            format: texheader.format.to_wgpu()?,
            width: texheader.width,
            height: texheader.height,
            depth: texheader.depth,
            array_size: texheader.array_size,
            large_buffer: texheader.large_buffer,

            deswizzle: Some((texheader.flags1 & 0xc00) != 0x400),
            psformat: Some(texheader.format),
        }
    }
}
```